File: Lista Tarefas/Lista Tarefas/Lista Tarefas/ViewController.m
//
//  ViewController.m
//  Lista Tarefas
//
//  Created by Kaique de Souza Monteiro on 17/03/15.
//  Copyright (c) 2015 Kaique de Souza Monteiro. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITableView *tasksTable;
@property (nonatomic) NSMutableArray* tasks;
@property (weak, nonatomic) IBOutlet UITextField *task;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _tasks = [[NSMutableArray alloc] init];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)addClick:(id)sender {
    
    NSString *task = _task.text;
    [_tasks addObject:task];
    _task.text = @"";
    [_tasksTable reloadData];
    
}

- (NSInteger)tableView:(UITableView *)tableView
numberOfRowsInSection:(NSInteger)section

{
    return [_tasks count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView
cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    static NSString *cellIdentifier;
    UITableViewCell *cell;
    
    
    cellIdentifier = @"SimpleTableCell";
    cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    
    
    if (cell == nil)
    {
        
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault
                reuseIdentifier:cellIdentifier];
        
    }
    
    cell.textLabel.text = [_tasks objectAtIndex:indexPath.row];
    
    
    
    return cell;
    
}

@end

File: Cinco Bola/Cinco Bola/Cinco Bola/ViewController.swift
//
//  ViewController.swift
//  Cinco Bola
//
//  Created by Kaique de Souza Monteiro on 13/04/15.
//  Copyright (c) 2015 Kaique de Souza Monteiro. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var firstTable: UITableView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func tableView(tableView: UITableView!, numberOfRowsInSection section: Int) -> Int {
        return 3
    }
    
    func tableView(tableView: UITableView!, cellForRowAtIndexPath indexPath: NSIndexPath!) -> UITableViewCell! {
        var cell  = tableView.dequeueReusableCellWithIdentifier("testsIndentifier") as? Cell
        
        if (cell == nil) {
            cell = Cell(style: UITableViewCellStyle.Value1, reuseIdentifier: "testsIndentifier")
        }
        
        return changeCellContend(cell!, index: indexPath.row)
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        var cell = tableView.cellForRowAtIndexPath(indexPath) as? Cell
        cell?.valueField.becomeFirstResponder()
        cell?.selectionStyle = UITableViewCellSelectionStyle.None
        
    }


    func changeCellContend (cell: Cell, index: Int) ->Cell {
        if(index == 2) {
            cell.descriptionLabel.text = "Peso da ultima prova"
        } else if(index == 1) {
            cell.descriptionLabel.text = "Número de provas"
        }
        return cell
    }

}

